### ISSUE TITLE
Fix `:focus` styles on `/notifications` for keyboard users

### ISSUE THREAD

**Issue by PhilipHow** [forem/forem#19685]
**Describe the bug**

The `:focus` styles on the "Like" and "Subscribe to Comments" buttons on `/notifications' are unclear for keyboard users.

![Image](https://github.com/forem/forem/assets/1628871/0c61d76d-6107-47ae-b3b1-eb85fe4e2d50)

![Image](https://github.com/forem/forem/assets/1628871/e3a4a960-420c-47c3-a4d1-887fc92a2478)

Compared to the nice outline state as shown here:

![Image](https://github.com/forem/forem/assets/1628871/6b04792b-e3d4-471e-9e1f-a48749ba1adf)

**To Reproduce**

1. Press tab on the `/notifications` page

**Expected behavior**

- We should match the existing style of the other buttons on the page
- We should look to reduce the impact to mouse users (so the focus style is not visible when using the mouse).
  - One approach is detailed [here](https://twitter.com/LeaVerou/status/1045768279753666562), which seems now to have decent browser support. 
  - Another option is to remove focus on click, but I'm unsure of any a11y concerns with this approach.

**Desktop:**

- OS, version: Windows 11 Pro
- Browser, version: Google Chrome Version 114.0.5735.199 (Official Build) (64-bit)

**Additional context**

- Related to #19555
